lib/install/actions.js

```diff
@@ -46,7 +46,8 @@ Object.keys(actions).forEach(function (actionName) {
       if (!pkg.commit) pkg.commit = []
       pkg.commit.push(action.commit)
     }
-    if (pkg.knownInstallable) {
+
+    if (pkg.knownInstallable || pkg.package._optional) {
```

Comment:
My reading of this code:

If the shrinkwrap flagged this package as optional then don't bother checking to see if it meets archictecture or OS requirements?

Isn't that the opposite of what we want?